createPackage.ps1

```diff
@@ -1,13 +1,57 @@
-$version = $env:version
-$revision = $env:revision
-$path = "gocd-" + $args[0]
-$fullVersion = $version + "-" + $revision
-$type = $args[0]
-cd $path
-$checksumFileContent = (New-Object System.Net.WebClient).DownloadString("https://download.gocd.io/binaries/$fullVersion/win/go-$type-$fullVersion-jre-32bit-setup.exe.sha256sum")
-$checksum32 = $checksumFileContent.Split(" ")[0]
-$checksumFileContent = (New-Object System.Net.WebClient).DownloadString("https://download.gocd.io/binaries/$fullVersion/win/go-$type-$fullVersion-jre-64bit-setup.exe.sha256sum")
-$checksum64 = $checksumFileContent.Split(" ")[0]
+<#
+    .SYNOPSIS
+        Build NuGet Packages (.nupkg) for GoCD Server or Agent.
+
+    .PARAMETER type
+        The type of package being built--either agent or server.
+
+    .PARAMETER version
+        The release version of the package being built (ie, YY.MM.XX).
+        Defaults to $env:version.
+
+    .PARAMETER revision
+        The build number of the package being built.
+        Defaults to $env:revision
+
+    .EXAMPLE
+        Using environment variables.
+
+        $env:version=17.4.0
+        $env:revision=4892
+        .\createPackage.ps1 server
+        .\createPackage.ps1 agent
+
+    .EXAMPLE
+        Using unnamed arguments.
+
+        .\createPackage.ps1 server 17.4.0 4892
+        .\createPackage.ps1 agent 17.4.0 4892
+
+    .EXAMPLE
+        Using named arguments.
+
+        .\createPackage.ps1 -type server -version 17.4.0 -revision 4892
+        .\createPackage.ps1 -type agent -version 17.4.0 -revision 4892
+#>
+param(
+    [string][parameter(mandatory=$true)][ValidateSet("agent", "server")]$type,
+
+    # Note that parameter validation doesn't work for parameter defaults,
+    # so if $env:version or $env:revision are unset, things will probably
+    # fail in some undefined way.
+    [string][parameter()][ValidateNotNullOrEmpty()]$version = $env:version,
+    [string][parameter()][ValidateNotNullOrEmpty()]$revision = $env:revision
+)
+
+$path = "gocd-$type"
+$fullVersion = "$version-$revision"
+
+Push-Location $path
+
+$checksum32 = (Invoke-RestMethod "https://download.gocd.io/binaries/$fullVersion/win/go-$type-$fullVersion-jre-32bit-setup.exe.sha256sum").Split(' ')[0]
```

Comment:
Hey, the reason I chose `System.Net.WebClient` was that we are using powershell 2.0 to build these packages for pushing. `Invoke-RestMethod` won't be available. I don't see any reason to update the version of powershell on the machines we used to build the packages because we are not losing out of any features for building the packages that poweshell v3 offers and not 2. Can you please change this back to using the WebClient?